eessi/reframe/eessi_utils/utils.py

```diff
@@ -3,24 +3,33 @@
 """
 
 import re
+from typing import Iterator
 
 import reframe as rfm
 import reframe.core.runtime as rt
 from reframe.utility import OrderedSet
 
 GPU_DEV_NAME = 'gpu'
 
-SCALES = [
-    # (scale_tag, nodes)
-    ('1_node', 1),
-    ('2_nodes', 2),
-    ('4_nodes', 4),
-    ('8_nodes', 8),
-    ('16_nodes', 16),
-]
-
-
-def _get_gpu_list(test: rfm.RegressionTest):
+SCALES = {
```

Comment:
The one thing I am concerned about is that supporting the `node_part` options will require us to write more logic in tests / hooks to make sure they work for _all_ these `SCALES` definitions. But let's see how it goes.

It _is_ clearly the most sensible way to allocate e.g. a single GPU in a 4-GPU node, for which you don't a priori know the CPU core count. E.g. for us a quarter GPU node would be 18 CPU cores + 1 GPU, I'm sure it's different for you.

So, let's just see how it goes. As long as the management of the logic doesn't become too crazy, let's keep it :)